services/clusters-keeper/src/simcore_service_clusters_keeper/modules/ec2.py

```diff
@@ -0,0 +1,259 @@
+import contextlib
+import logging
+from dataclasses import dataclass
+from typing import cast
+
+import aioboto3
+import botocore.exceptions
+from aiobotocore.session import ClientCreatorContext
+from fastapi import FastAPI
+from pydantic import ByteSize, parse_obj_as
+from servicelib.logging_utils import log_context
+from tenacity._asyncio import AsyncRetrying
+from tenacity.before_sleep import before_sleep_log
+from tenacity.stop import stop_after_delay
+from tenacity.wait import wait_random_exponential
+from types_aiobotocore_ec2 import EC2Client
+from types_aiobotocore_ec2.literals import InstanceStateNameType, InstanceTypeType
+from types_aiobotocore_ec2.type_defs import FilterTypeDef
+
+from ..core.errors import (
+    ConfigurationError,
+    Ec2InstanceNotFoundError,
+    Ec2NotConnectedError,
+    Ec2TooManyInstancesError,
+)
+from ..core.settings import EC2InstancesSettings, EC2Settings, get_application_settings
+from ..models import EC2InstanceData, EC2InstanceType
+from ..utils.ec2 import compose_user_data
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ClustersKeeperEC2:
+    client: EC2Client
+    session: aioboto3.Session
+    exit_stack: contextlib.AsyncExitStack
+
+    @classmethod
+    async def create(cls, settings: EC2Settings) -> "ClustersKeeperEC2":
+        session = aioboto3.Session()
+        session_client = session.client(
+            "ec2",
+            endpoint_url=settings.EC2_ENDPOINT,
+            aws_access_key_id=settings.EC2_ACCESS_KEY_ID,
+            aws_secret_access_key=settings.EC2_SECRET_ACCESS_KEY,
+            region_name=settings.EC2_REGION_NAME,
+        )
+        assert isinstance(session_client, ClientCreatorContext)  # nosec
+        exit_stack = contextlib.AsyncExitStack()
+        ec2_client = cast(
+            EC2Client, await exit_stack.enter_async_context(session_client)
+        )
+        return cls(ec2_client, session, exit_stack)
+
+    async def close(self) -> None:
+        await self.exit_stack.aclose()
+
+    async def ping(self) -> bool:
+        try:
+            await self.client.describe_account_attributes()
+            return True
+        except Exception:  # pylint: disable=broad-except
+            return False
```

Comment:
I would rename this method, In my head it was used to ping the EC2 instance.

Reply:
EC2 is an AWS service. That is what is being pinged here.